cli/src/semgrep/console_scripts/entrypoint.py

```diff
@@ -178,20 +179,24 @@ def exec_osemgrep():
 
     # If you call opengrep-core as opengrep-cli, then we get
     # opengrep-cli behavior, see src/main/Main.ml
-    sys.argv[0] = "opengrep-cli" 
-
-    # nosem: dangerous-os-exec-tainted-env-args
-    os.execvp(str(path), sys.argv)
+    sys.argv[0] = "opengrep-cli"
 
+    if IS_WINDOWS:
+      cp = subprocess.run(sys.argv, executable=str(path), close_fds=True)
+      sys.exit(cp.returncode)
+    else:
+      # nosem: dangerous-os-exec-tainted-env-args
+      os.execvp(str(path), sys.argv)
 
 # Needed for similar reasons as in pysemgrep, but only for the legacy
 # flag to work
 def main():
 
     if getattr(sys, 'frozen', False) and hasattr(sys, '_MEIPASS'):  # PyInstaller
         os.environ["_OPENGREP_BINARY"] = sys.executable
-        if IS_WINDOWS: # Because `execvp` on windows does not replace the current process.
-            os.environ["PYINSTALLER_RESET_ENVIRONMENT"] = "1"
+        # NOTE: Not needed because we now use `subprocess.run` and we wait for completion.
+        # if IS_WINDOWS: # Because `execvp` on windows does not replace the current process.
```

Comment:
?

Reply:
Comment to be removed on next PR, indeed it is no longer needed because we now do not allow the process to be killed which is what happens on Windows with execvp.